### ISSUE TITLE
Fix the error about get_lwp_regs()

### ISSUE THREAD

**Issue by shining1984** [openjdk-riscv/jdk11u#119]
In file included from /home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/LinuxDebuggerLocal.c:26:
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc.h:114:65: warning: 'struct user_regs_struct' declared inside parameter list will not be visible outside of this definition or declaration
  114 | bool get_lwp_regs(struct ps_prochandle* ph, lwpid_t lid, struct user_regs_struct* regs);
      |                                                                 ^~~~~~~~~~~~~~~~
In file included from /home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.h:30,
                 from /home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.c:30:
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc.h:114:65: warning: 'struct user_regs_struct' declared inside parameter list will not be visible outside of this definition or declaration
  114 | bool get_lwp_regs(struct ps_prochandle* ph, lwpid_t lid, struct user_regs_struct* regs);
      |                                                                 ^~~~~~~~~~~~~~~~
In file included from /home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.c:30:
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.h:50:29: error: field 'regs' has incomplete type
   50 |    struct user_regs_struct  regs;       // not for process, core uses for caching regset
      |                             ^~~~
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.c:376:6: error: conflicting types for 'get_lwp_regs'
  376 | bool get_lwp_regs(struct ps_prochandle* ph, lwpid_t lwp_id, struct user_regs_struct* regs) {
      |      ^~~~~~~~~~~~
In file included from /home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.h:30,
                 from /home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc_impl.c:30:
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.hotspot.agent/linux/native/libsaproc/libproc.h:114:6: note: previous declaration of 'get_lwp_regs' was here
  114 | bool get_lwp_regs(struct ps_prochandle* ph, lwpid_t lid, struct user_regs_struct* regs);
      |      ^~~~~~~~~~~~
Lib-jdk.hotspot.agent.gmk:58: recipe for target '/home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/support/native/jdk.hotspot.agent/libsaproc/libproc_impl.o' failed
make[3]: *** [/home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/support/native/jdk.hotspot.agent/libsaproc/libproc_impl.o] Error 1
make[2]: *** [jdk.hotspot.agent-libs] Error 2
make/Main.gmk:215: recipe for target 'jdk.hotspot.agent-libs' failed

ERROR: Build failed for target 'default (exploded-image)' in configuration 'linux-riscv32-normal-core-fastdebug' (exit code 2) 
Stopping sjavac server

**Comment by shining1984** [comment]
[log.txt](https://github.com/openjdk-riscv/jdk11u/files/6362180/log.txt)
完整的log ，来自issue #116 。

**Comment by shining1984** [comment]
为src/jdk.hotspot.agent/linux/native/libsaproc/libproc.h添加 riscv32 define判断。
添加后，可以保证编译通过。

**Comment by shining1984** [comment]
shining@shining-VirtualBox:~/jdk-dev/shiningning/jdk11u$ bash configure --with-boot-jdk=/home/shining/jdk-10 --openjdk-target=riscv32-unknown-linux-gnu --with-sysroot=/home/shining/riscv32/sysroot --disable-warnings-as-errors --with-debug-level=fastdebug --with-native-debug-symbols=internal --with-jvm-variants=core --prefix=$PWD/fastdebug_32_core
configure: Configuration created at Fri Apr 23 10:32:36 CST 2021.
checking for basename... /usr/bin/basename
checking for bash... /bin/bash
checking for cat... /bin/cat
checking for chmod... /bin/chmod
checking for cmp... /usr/bin/cmp
checking for comm... /usr/bin/comm
checking for cp... /bin/cp
checking for cut... /usr/bin/cut
checking for date... /bin/date
checking for gdiff... no
checking for diff... /usr/bin/diff
checking for dirname... /usr/bin/dirname
checking for echo... /bin/echo
checking for expr... /usr/bin/expr
checking for file... /usr/bin/file
checking for find... /usr/bin/find
checking for head... /usr/bin/head
checking for gunzip... /bin/gunzip
checking for pigz... no
checking for gzip... /bin/gzip
checking for ln... /bin/ln
checking for ls... /bin/ls
checking for mkdir... /bin/mkdir
checking for mktemp... /bin/mktemp
checking for mv... /bin/mv
checking for nawk... /usr/bin/nawk
checking for printf... /usr/bin/printf
checking for greadlink... no
checking for readlink... /bin/readlink
checking for rm... /bin/rm
checking for rmdir... /bin/rmdir
checking for sh... /bin/sh
checking for sort... /usr/bin/sort
checking for tail... /usr/bin/tail
checking for gtar... no
checking for tar... /bin/tar
checking for tee... /usr/bin/tee
checking for touch... /usr/bin/touch
checking for tr... /usr/bin/tr
checking for uname... /bin/uname
checking for uniq... /usr/bin/uniq
checking for wc... /usr/bin/wc
checking for which... /usr/bin/which
checking for xargs... /usr/bin/xargs
checking for gawk... gawk
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for a sed that does not truncate output... /bin/sed
checking for cygpath... no
checking for df... /bin/df
checking for cpio... /bin/cpio
checking for nice... /usr/bin/nice
checking for pandoc... no
checking build system type... x86_64-unknown-linux-gnu
checking host system type... riscv32-unknown-linux-gnu
checking target system type... riscv32-unknown-linux-gnu
checking openjdk-build os-cpu... linux-x86_64
checking openjdk-target os-cpu... linux-riscv32
checking compilation type... cross
checking for top-level directory... /home/shining/jdk-dev/shiningning/jdk11u
checking if custom source is suppressed (openjdk-only)... no
checking which variant of the JDK to build... normal
checking which debug level to use... fastdebug
checking which variants of the JVM to build... core
checking if absolute paths should be allowed in the build output... yes, debug build
checking for sysroot... /home/shining/riscv32/sysroot
checking for toolchain path... 
checking for extra path... 
checking where to store configuration... in default location
checking what configuration name to use... linux-riscv32-normal-core-fastdebug
checking for apt-get... apt-get
checking for gmake... no
checking for make... /usr/bin/make
configure: Testing potential make at /usr/bin/make, found using make in PATH
configure: Using GNU make at /usr/bin/make (version: GNU Make 4.1)
checking if make --output-sync is supported... yes
checking for output-sync value... none
checking if find supports -delete... yes
checking what type of tar was found... gnu
checking that grep (/bin/grep) -Fx handles empty lines in the pattern list correctly... yes
checking for unzip... /usr/bin/unzip
checking for zip... /usr/bin/zip
checking for ldd... /usr/bin/ldd
checking for greadelf... no
checking for readelf... /usr/bin/readelf
checking for dot... no
checking for hg... /usr/bin/hg
checking for git... /usr/bin/git
checking for stat... /usr/bin/stat
checking for time... /usr/bin/time
checking for flock... /usr/bin/flock
checking for dtrace... no
checking for gpatch... no
checking for patch... /usr/bin/patch
checking bash version... 4.4.20
checking if bash supports pipefail... yes
checking if bash supports errexit (-e)... yes
configure: WARNING: using cross tools not prefixed with host triplet
checking for riscv32-unknown-linux-gnu-pkg-config... no
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for default LOG value... 
checking headless only... no
checking for graphviz dot... no, cannot generate full docs
checking for pandoc... no, cannot generate full docs
checking full docs... no, missing dependencies
checking for cacerts file... default
checking if packaged modules are kept... yes (default)
checking for version string... 11.0.9-internal+0-adhoc.shining.jdk11u
configure: Found potential Boot JDK using configure arguments
checking for Boot JDK... /home/shining/jdk-10
checking Boot JDK version... openjdk version "10" 2018-03-20 OpenJDK Runtime Environment 18.3 (build 10+44) OpenJDK 64-Bit Server VM 18.3 (build 10+44, mixed mode) 
checking for java in Boot JDK... ok
checking for javac in Boot JDK... ok
checking for javadoc in Boot JDK... ok
checking for jar in Boot JDK... ok
checking for jarsigner in Boot JDK... ok
checking if Boot JDK is 32 or 64 bits... 64
checking for local Boot JDK Class Data Sharing (CDS)... yes, created
checking for Build JDK... yes, will build it for the host platform
configure: Using default toolchain gcc (GNU Compiler Collection)
checking for riscv32-unknown-linux-gnu-gcc... /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc
checking resolved symbolic links for CC... no symlink
configure: Using gcc C compiler version 10.2.0 [riscv32-unknown-linux-gnu-gcc (GCC) 10.2.0]
checking for riscv32-unknown-linux-gnu-/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc... /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc accepts -g... yes
checking for /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc option to accept ISO C89... none needed
checking for riscv32-unknown-linux-gnu-g++... /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-g++
checking resolved symbolic links for CXX... no symlink
configure: Using gcc C++ compiler version 10.2.0 [riscv32-unknown-linux-gnu-g++ (GCC) 10.2.0]
checking whether we are using the GNU C++ compiler... yes
checking whether /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-g++ accepts -g... yes
checking how to run the C preprocessor... /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc -E
checking how to run the C++ preprocessor... /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-g++ -E
checking for riscv32-unknown-linux-gnu-ld... riscv32-unknown-linux-gnu-ld
configure: Rewriting LD_JAOTC to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-ld"
configure: Using gcc linker version 2.35 [GNU ld (GNU Binutils) 2.35]
checking for riscv32-unknown-linux-gnu-ar... riscv32-unknown-linux-gnu-ar
configure: Rewriting AR to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-ar"
checking for riscv32-unknown-linux-gnu-strip... riscv32-unknown-linux-gnu-strip
configure: Rewriting STRIP to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-strip"
checking for riscv32-unknown-linux-gnu-nm... riscv32-unknown-linux-gnu-nm
configure: Rewriting NM to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-nm"
checking for riscv32-unknown-linux-gnu-gobjcopy... no
checking for riscv32-unknown-linux-gnu-objcopy... riscv32-unknown-linux-gnu-objcopy
configure: Rewriting OBJCOPY to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-objcopy"
checking for riscv32-unknown-linux-gnu-gobjdump... no
checking for riscv32-unknown-linux-gnu-objdump... riscv32-unknown-linux-gnu-objdump
configure: Rewriting OBJDUMP to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-objdump"
checking for riscv32-unknown-linux-gnu-c++filt... riscv32-unknown-linux-gnu-c++filt
configure: Rewriting CXXFILT to "/home/shining/riscv32/bin/riscv32-unknown-linux-gnu-c++filt"
checking for cl... no
checking for cc... /usr/bin/cc
checking for cl... no
checking for CC... no
checking for g++... /usr/bin/g++
checking for nm... /usr/bin/nm
checking for ar... /usr/bin/ar
checking for objcopy... /usr/bin/objcopy
checking for strip... /usr/bin/strip
configure: Using gcc BuildC compiler version 7.5.0 [cc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0]
configure: Using gcc BuildC++ compiler version 7.5.0 [g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0]
configure: Using gcc build linker version 2.35 [GNU ld (GNU Binutils) 2.35]
checking for jtreg... no
checking for jtreg test harness... no, not found
checking for jib... no
checking if @file is supported by gcc... yes
checking if CC supports "-m32"... no
checking if CXX supports "-m32"... no
checking if both CC and CXX support "-m32"... no
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking size of int *... 4
checking for target address size... 32 bits
checking whether byte ordering is bigendian... no
checking if native warnings are errors... no (explicitly set)
checking if CC supports "-Xassembler -mrelax-relocations=no"... no
checking if CXX supports "-Xassembler -mrelax-relocations=no"... no
checking if both CC and CXX support "-Xassembler -mrelax-relocations=no"... no
checking for library containing clock_gettime... none required
checking if CXX supports "-std=gnu++98 -Werror"... yes
checking if CC supports "-fno-delete-null-pointer-checks -Werror"... yes
checking if CXX supports "-fno-delete-null-pointer-checks -Werror"... yes
checking if both CC and CXX support "-fno-delete-null-pointer-checks -Werror"... yes
checking if CC supports "-fno-lifetime-dse -Werror"... yes
checking if CXX supports "-fno-lifetime-dse -Werror"... yes
checking if both CC and CXX support "-fno-lifetime-dse -Werror"... yes
checking if CC supports "-ffp-contract=off"... yes
checking if CXX supports "-ffp-contract=off"... yes
checking if both CC and CXX support "-ffp-contract=off"... yes
checking if BUILD_CXX supports "-std=gnu++98 -Werror"... yes
checking if BUILD_CC supports "-fno-delete-null-pointer-checks -Werror"... yes
checking if BUILD_CXX supports "-fno-delete-null-pointer-checks -Werror"... yes
checking if both BUILD_CC and BUILD_CXX support "-fno-delete-null-pointer-checks -Werror"... yes
checking if BUILD_CC supports "-fno-lifetime-dse -Werror"... yes
checking if BUILD_CXX supports "-fno-lifetime-dse -Werror"... yes
checking if both BUILD_CC and BUILD_CXX support "-fno-lifetime-dse -Werror"... yes
checking if BUILD_CC supports "-ffp-contract=off"... yes
checking if BUILD_CXX supports "-ffp-contract=off"... yes
checking if both BUILD_CC and BUILD_CXX support "-ffp-contract=off"... yes
checking what type of native debug symbols to use... internal
checking if we should add external native debug symbols to the shipped bundles... no
checking for dtrace tool... not found, cannot build dtrace
checking sys/sdt.h usability... no
checking sys/sdt.h presence... no
checking for sys/sdt.h... no
checking if dtrace should be built... no, missing dependencies
checking if Hotspot gtest unit tests should be built... yes
checking if static link of stdc++ is possible... yes
checking how to link with libstdc++... static
checking for X... libraries /home/shining/riscv32/sysroot/usr/lib, headers /home/shining/riscv32/sysroot/usr/include
checking whether -R must be followed by a space... neither works
checking for gethostbyname... yes
checking for connect... yes
checking for remove... yes
checking for shmat... yes
checking for IceConnectionNumber in -lICE... yes
checking for X11/extensions/shape.h... yes
checking for X11/extensions/Xrender.h... yes
checking for X11/extensions/XTest.h... yes
checking for X11/Intrinsic.h... yes
checking for X11/extensions/Xrandr.h... yes
checking if XlinearGradient is defined in Xrender.h... yes
checking cups/cups.h usability... yes
checking cups/cups.h presence... yes
checking for cups/cups.h... yes
checking cups/ppd.h usability... yes
checking cups/ppd.h presence... yes
checking for cups/ppd.h... yes
checking fontconfig/fontconfig.h usability... yes
checking fontconfig/fontconfig.h presence... yes
checking for fontconfig/fontconfig.h... yes
configure: Found freetype include files at /home/shining/riscv32/sysroot/usr/include/freetype2 using well-known location
checking for freetype includes... /home/shining/riscv32/sysroot/usr/include/freetype2
checking for freetype libraries... /home/shining/riscv32/sysroot/usr/lib
Using freetype: system
checking alsa/asoundlib.h usability... yes
checking alsa/asoundlib.h presence... yes
checking for alsa/asoundlib.h... yes
checking for which libjpeg to use... bundled
checking for which giflib to use... bundled
checking for PNG... yes
checking for which libpng to use... bundled
checking for compress in -lz... yes
checking for which zlib to use... system
checking for system zlib functionality... ok
checking for which lcms to use... bundled
checking for cos in -lm... yes
checking for dlopen in -ldl... yes
checking if shenandoah can be built... checking if zgc can be built... no, platform not supported
checking if jvmci module jdk.internal.vm.ci should be built... no
checking if graal module jdk.internal.vm.compiler should be built... no
checking if aot should be enabled... no
checking if cds should be enabled... yes
checking if elliptic curve crypto implementation is present... yes
checking if jtreg failure handler should be built... no, missing jtreg
checking if the CDS classlist generation should be enabled... no
checking if any translations should be excluded... no
checking if man pages should be built... yes
checking for number of cores... 1
checking for memory size... 9679 MB
checking for appropriate number of jobs to run in parallel... 1
checking flags for boot jdk java command ...  -Duser.language=en -Duser.country=US  -XX:+UnlockDiagnosticVMOptions -XX:-VerifySharedSpaces -XX:SharedArchiveFile=/home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/configure-support/classes.jsa -Xshare:auto 
checking flags for boot jdk java command for big workloads...  -Xms64M -Xmx1600M -XX:ThreadStackSize=1536
checking flags for bootcycle boot jdk java command for big workloads... -Xms64M -Xmx768M -XX:ThreadStackSize=768
checking flags for boot jdk java command for small workloads...  -XX:+UseSerialGC -Xms32M -Xmx512M -XX:TieredStopAtLevel=1
checking whether to use sjavac... no
checking whether to use javac server... yes
checking If precompiled header is enabled... yes
checking that precompiled headers work... yes
checking is ccache enabled... no
checking if build directory is on local disk... yes
checking JVM features for JVM variant 'core'... "cds cmsgc epsilongc g1gc jfr jni-check jvmti management nmt parallelgc serialgc services vm-structs"
configure: creating /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/configure-support/config.status
config.status: creating /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/spec.gmk
config.status: creating /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/bootcycle-spec.gmk
config.status: creating /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/buildjdk-spec.gmk
config.status: creating /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/compare.sh
config.status: creating /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/Makefile

====================================================
The existing configuration has been successfully updated in
/home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug
using configure arguments '--with-boot-jdk=/home/shining/jdk-10 --openjdk-target=riscv32-unknown-linux-gnu --with-sysroot=/home/shining/riscv32/sysroot --disable-warnings-as-errors --with-debug-level=fastdebug --with-native-debug-symbols=internal --with-jvm-variants=core --prefix=/home/shining/jdk-dev/shiningning/jdk11u/fastdebug_32_core'.

Configuration summary:
* Debug level:    fastdebug
* HS debug level: fastdebug
* JVM variants:   core
* JVM features:   core: 'cds cmsgc epsilongc g1gc jfr jni-check jvmti management nmt parallelgc serialgc services vm-structs' 
* OpenJDK target: OS: linux, CPU architecture: riscv32, address length: 32
* Version string: 11.0.9-internal+0-adhoc.shining.jdk11u (11.0.9-internal)

Tools summary:
* Boot JDK:       openjdk version "10" 2018-03-20 OpenJDK Runtime Environment 18.3 (build 10+44) OpenJDK 64-Bit Server VM 18.3 (build 10+44, mixed mode)  (at /home/shining/jdk-10)
* Toolchain:      gcc (GNU Compiler Collection)
* C Compiler:     Version 10.2.0 (at /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-gcc)
* C++ Compiler:   Version 10.2.0 (at /home/shining/riscv32/bin/riscv32-unknown-linux-gnu-g++)

Build performance summary:
* Cores to use:   1
* Memory limit:   9679 MB

WARNING: The result of this configuration has overridden an older
configuration. You *should* run 'make clean' to make sure you get a
proper build. Failure to do so might result in strange build problems.

The following warnings were produced. Repeated here for convenience:
WARNING: using cross tools not prefixed with host triplet

shining@shining-VirtualBox:~/jdk-dev/shiningning/jdk11u$ make -j
Building target 'default (exploded-image)' in configuration 'linux-riscv32-normal-core-fastdebug'
Updating support/modules_libs/java.base/server/libjvm.so due to 1 file(s)
Creating support/modules_cmds/jdk.hotspot.agent/jhsdb from 1 file(s)
Compiling 44 files for jdk.httpserver
Compiling 97 files for jdk.internal.le
Compiling 51 files for jdk.internal.opt
Compiling 11 properties into resource bundles for jdk.jartool
Compiling 33 files for jdk.jartool
Creating support/modules_cmds/jdk.jartool/jar from 1 file(s)
Creating support/modules_cmds/jdk.jartool/jarsigner from 1 file(s)
Compiling 300 files for jdk.javadoc
Creating support/modules_cmds/jdk.javadoc/javadoc from 1 file(s)
Compiling 40 files for jdk.jcmd
Creating support/modules_cmds/jdk.jcmd/jinfo from 1 file(s)
Creating support/modules_cmds/jdk.jcmd/jmap from 1 file(s)
Creating support/modules_cmds/jdk.jcmd/jps from 1 file(s)
Creating support/modules_cmds/jdk.jcmd/jstack from 1 file(s)
Creating support/modules_cmds/jdk.jcmd/jstat from 1 file(s)
Creating support/modules_cmds/jdk.jcmd/jcmd from 1 file(s)
Compiling 24 files for jdk.management
Compiling 11 properties into resource bundles for jdk.management.agent
Compiling 31 files for jdk.management.agent
Compiling 64 files for jdk.jconsole
Creating support/modules_cmds/jdk.jconsole/jconsole from 1 file(s)
Compiling 12 properties into resource bundles for jdk.jdeps
Compiling 133 files for jdk.jdeps
Creating support/modules_cmds/jdk.jdeps/javap from 1 file(s)
Creating support/modules_cmds/jdk.jdeps/jdeps from 1 file(s)
Creating support/modules_cmds/jdk.jdeps/jdeprscan from 1 file(s)
Compiling 3 properties into resource bundles for jdk.jdi
Compiling 1 files for jdk.jdwp.agent
Compiling 251 files for jdk.jdi
Creating support/modules_cmds/jdk.jdi/jdb from 1 file(s)
Creating support/modules_libs/jdk.jdwp.agent/libdt_socket.so from 2 file(s)
Creating support/modules_libs/jdk.jdwp.agent/libjdwp.so from 42 file(s)
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.jdwp.agent/share/native/libjdwp/log_messages.c: In function 'log_message_end':
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.jdwp.agent/share/native/libjdwp/log_messages.c:75:24: warning: '%.3d' directive output may be truncated writing between 3 and 11 bytes into a region of size between 0 and 80 [-Wformat-truncation=]
   75 |                    "%s.%.3d %s", timestamp_prefix,
      |                        ^~~~
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.jdwp.agent/share/native/libjdwp/log_messages.c:75:20: note: using the range [-2147483648, 2147483647] for directive argument
   75 |                    "%s.%.3d %s", timestamp_prefix,
      |                    ^~~~~~~~~~~~
/home/shining/jdk-dev/shiningning/jdk11u/src/jdk.jdwp.agent/share/native/libjdwp/log_messages.c:74:11: note: 'snprintf' output between 6 and 174 bytes into a destination of size 81
   74 |     (void)snprintf(tbuf, ltbuf,
      |           ^~~~~~~~~~~~~~~~~~~~~
   75 |                    "%s.%.3d %s", timestamp_prefix,
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   76 |                    (int)(millisecs), timestamp_postfix);
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiling 175 files for jdk.jfr
Creating support/modules_cmds/jdk.jfr/jfr from 1 file(s)
Compiling 4 properties into resource bundles for jdk.jlink
Compiling 3 properties into resource bundles for jdk.jlink
Compiling 1 properties into resource bundles for jdk.jlink
Compiling 81 files for jdk.jlink
Creating support/modules_cmds/jdk.jlink/jimage from 1 file(s)
Creating support/modules_cmds/jdk.jlink/jlink from 1 file(s)
Creating support/modules_cmds/jdk.jlink/jmod from 1 file(s)
Compiling 7 properties into resource bundles for jdk.jshell
Compiling 94 files for jdk.jshell
Creating support/modules_cmds/jdk.jshell/jshell from 1 file(s)
Compiling 5 files for jdk.jsobject
Compiling 11 files for jdk.jstatd
Creating support/modules_cmds/jdk.jstatd/jstatd from 1 file(s)
Compiling 224 properties into resource bundles for jdk.localedata
Compiling 1703 files for jdk.localedata
Creating support/modules_libs/jdk.management/libmanagement_ext.so from 8 file(s)
Creating support/modules_libs/jdk.management.agent/libmanagement_agent.so from 1 file(s)
Compiling 14 files for jdk.management.jfr
Compiling 16 files for jdk.naming.dns
Compiling 4 files for jdk.naming.ldap
Compiling 7 files for jdk.naming.rmi
Warning: generation and use of skeletons and static stubs for JRMP
is deprecated. Skeletons are unnecessary, and static stubs have
been superseded by dynamically generated stubs. Users are
encouraged to migrate away from using rmic to generate skeletons and static
stubs. See the documentation for java.rmi.server.UnicastRemoteObject.
Compiling 9 files for jdk.net
Creating support/modules_libs/jdk.net/libextnet.so from 1 file(s)
Compiling 1 files for jdk.pack
Creating support/modules_libs/jdk.pack/libunpack.so from 7 file(s)
Creating support/modules_cmds/jdk.pack/pack200 from 1 file(s)
Creating support/modules_cmds/jdk.pack/unpack200 from 7 file(s)
Compiling 188 files for jdk.rmic
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Creating support/modules_cmds/jdk.rmic/rmic from 1 file(s)
Compiling 593 files for jdk.scripting.nashorn
Note: Some input files use or override a deprecated API that is marked for removal.
Note: Recompile with -Xlint:removal for details.
Compiling 131 files for BUILD_NASGEN
Running nasgen
Compiling 11 files for jdk.scripting.nashorn.shell
Note: Some input files use or override a deprecated API that is marked for removal.
Note: Recompile with -Xlint:removal for details.
Creating support/modules_cmds/jdk.scripting.nashorn.shell/jjs from 1 file(s)
Compiling 33 files for jdk.sctp
Creating support/modules_libs/jdk.sctp/libsctp.so from 3 file(s)
Compiling 30 files for jdk.security.auth
Creating support/modules_libs/jdk.security.auth/libjaas.so from 1 file(s)
Compiling 16 files for jdk.security.jgss
Compiling 9 files for jdk.unsupported
Compiling 8 files for jdk.unsupported.desktop
Compiling 94 files for jdk.xml.dom
Compiling 15 files for jdk.zipfs
Stopping sjavac server
shining@shining-VirtualBox:~/jdk-dev/shiningning/jdk11u$ Finished building target 'default (exploded-image)' in configuration 'linux-riscv32-normal-core-fastdebug'

**Comment by shining1984** [comment]
@DingliZhang ， PR #120 ，PR已经发起，请review和验证。

**Comment by shining1984** [comment]
shining@shining-VirtualBox:~/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/jdk/bin$ /home/shining/qemu/bin/qemu-riscv32 -L /home/shining/riscv32/sysroot ./java -version
# To suppress the following error report, specify this argument
# after -XX: or in .hotspotrc:  SuppressErrorAt=/macroAssembler_riscv32.cpp:2632
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (/home/shining/jdk-dev/shiningning/jdk11u/src/hotspot/cpu/riscv32/macroAssembler_riscv32.cpp:2632), pid=10499, tid=10501
#  assert(ReservedCodeCacheSize < 4*G) failed: branch out of range
#
# JRE version:  (11.0.9) (fastdebug build )
# Java VM: OpenJDK Core VM (fastdebug 11.0.9-internal+0-adhoc.shining.jdk11u, interpreted mode, sharing, serial gc, linux-riscv32)
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P %E" (or dumping to /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/jdk/bin/core.10499)
#
# An error report file with more information is saved as:
# /home/shining/jdk-dev/shiningning/jdk11u/build/linux-riscv32-normal-core-fastdebug/jdk/bin/hs_err_pid10499.log
#
#
Current thread is 10501
Dumping core ...
Aborted (core dumped)

**Comment by DingliZhang** [comment]
已验证和合并
